app/controllers/workorders_controller.rb

```diff
@@ -21,13 +23,14 @@ def new
 
   def create
     @workorder = Workorder.new workorder_params
+    @workorder.update(building_id: session[:current_building_id])
```

Comment:
Same thing here. Will the `update` method actually work here? If it does, it will save the object in the db and you probably don't need the `if @workorder.save` statement below.